src/spatial/nanoflann_internal.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef NANOFLANN_INTERNAL_H_
```

Comment:
For internal headers, let's prefix the header guard with the directory name, and let's not use a trailing `_`.

So `SPATIAL_NANOFLANN_INTERNAL_H`